flow/enhancers/neutron.go

```diff
@@ -35,48 +35,31 @@ import (
 type NeutronFlowEnhancer struct {
 	graph.DefaultGraphListener
 	Graph    *graph.Graph
-	tidCache *tidCache
+	tidCache *graph.MetadataIndexer
 }
 
 // Name return the Neutron enahancer name
 func (nfe *NeutronFlowEnhancer) Name() string {
 	return "Neutron"
 }
 
-func (nfe *NeutronFlowEnhancer) getNodeTID(mac string) string {
+func (nfe *NeutronFlowEnhancer) getNodeTID(mac string) (tid string) {
 	if packet.IsBroadcastMac(mac) || packet.IsMulticastMac(mac) {
 		return "*"
 	}
 
-	var ce *tidCacheEntry
-
-	if nfe.tidCache != nil {
-		var f bool
-		if ce, f = nfe.tidCache.get(mac); f {
-			return ce.tid
-		}
-	}
-
 	nfe.Graph.RLock()
 	defer nfe.Graph.RUnlock()
 
-	var tid string
-
-	// use the PeerIntfMAC metadata provided by the neutron probe. The interface used is the
-	// one attached to the VM interface.
-	intfs := nfe.Graph.GetNodes(graph.Metadata{"PeerIntfMAC": mac, "Manager": "neutron"})
-	if len(intfs) > 1 {
-		logging.GetLogger().Infof("NeutronFlowEnhancer found more than one interface with the PeerIntfMAC: %s", mac)
-	} else if len(intfs) == 1 {
-		tid, _ = intfs[0].GetFieldString("TID")
-	}
-
-	if nfe.tidCache != nil {
-		logging.GetLogger().Debugf("NeutronFlowEnhancer set cache %s: %s", mac, tid)
-		nfe.tidCache.set(ce, mac, tid)
+	nodes, _ := nfe.tidCache.Get(mac)
+	if len(nodes) > 0 {
+		if len(nodes) > 1 {
+			logging.GetLogger().Infof("NeutronFlowEnhancer found more than one interface with the PeerIntfMAC: %s", mac)
+		}
+		tid, _ = nodes[0].GetFieldString("TID")
```

Comment:
Previously we didn't return the TID if matching multiple MAC, with this version we take the first one. Not sure this is good as we are going to report something that could be wrong.

Reply:
Restored the original behaviour